content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md

```diff
@@ -0,0 +1,367 @@
+
+Here is the high-level overview architecture that shows what we want to achieve at the end of the day:
+
+![cloud_functions_reference_arch](/docs/images/cloud_functions_reference_arch.png)
+
+You can find all the related code and resources in [this repository](https://github.com/Dentrax/kubernetes-response-engine-based-on-gke-and-gcloudfunctions).
+
+## Prerequisites
+
+- gcloud 342.0.0
+- kubectl 1.20.5
+
+## Tutorial
+
+### Provision Google Kubernetes Engine (GKE) Cluster
+
+As the blog title said already, we need to create a [GKE cluster](https://cloud.google.com/kubernetes-engine) with workload identity enabled:
+
+```shell
+$ GOOGLE_PROJECT_ID=$(gcloud config get-value project)
+$ CLUSTER_NAME=falco-falcosidekick-demo
+$ gcloud container clusters create $CLUSTER_NAME --workload-pool ${GOOGLE_PROJECT_ID}.svc.id.goog
+$ gcloud container clusters get-credentials $CLUSTER_NAME
+```
+
+### Configuring IAM Service Accounts
+
+We need to create a new [Service Account](https://cloud.google.com/iam/docs/service-accounts) for target `$GOOGLE_PROJECT_ID` using IAM Binding [policies](https://cloud.google.com/iam/docs/policies) to get access our Cloud Function:
+```shell
+$ SA_ACCOUNT=falco-falcosidekick-sa
+$ gcloud iam service-accounts create $SA_ACCOUNT
+
+$ gcloud projects add-iam-policy-binding ${GOOGLE_PROJECT_ID} \
+--member="serviceAccount:${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com" \
+--role="roles/cloudfunctions.developer"
+
+$ gcloud projects add-iam-policy-binding ${GOOGLE_PROJECT_ID} \
+--member="serviceAccount:${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com" \
+--role="roles/cloudfunctions.invoker"
+```
+
+At the beginning, we already enabled [WorkloadIdentity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) feature for our GKE Cluster by setting `--workload-pool` flag. What we need to do here is we should add a `iam.workloadIdentityUser` role for the given Service Account. 
+
+```bash
+$ gcloud iam service-accounts add-iam-policy-binding \
+  --role roles/iam.workloadIdentityUser \
+  --member "serviceAccount:${GOOGLE_PROJECT_ID}.svc.id.goog[${FALCO_NAMESPACE}/falco-falcosidekick]" \
+  ${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com
+```
+
+We need to _annotate_ the `falco-falcosidekick` resource. So it can grant access for our Cluster. Set up the Falcosidekick SA to impersonate a GCP SA:
+```bash
+$ kubectl annotate serviceaccount \
+  --namespace $FALCO_NAMESPACE \
+  falco-falcosidekick \
+  iam.gke.io/gcp-service-account=${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com
+```
+
+### Create Necessary Cluster Role
+
+To limit function role access in the particular cluster, we need to ensure our SA only have limited permissions within a particular namespace by using [Role Bindings](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding). It should only access _Pod_ resource for _DELETE_ action:
```

Comment:
To limit function role access in the particular cluster, ensure that only SA has limited permissions within a particular namespace by using [Role Bindings](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding).